src/main/java/StudentSystem/Controller/CourseController.java

```diff
@@ -40,7 +45,15 @@ public String addCourse(@Valid @ModelAttribute("courseAdd") Course course, Bindi
         if (result.hasErrors()) {
             return "error";
         }
+        if ((courseRepository.existsById(course.getId()))) {
```

Comment:
Redundant wrapping.
Also, see if you can do it in a different method - a helper class of sorts.